src/index.js

```diff
@@ -1,31 +1,40 @@
 import * as babel from '@babel/core';
 import { createFilter } from 'rollup-pluginutils';
-import createPreflightCheck from './preflightCheck.js';
-import helperPlugin from './helperPlugin.js';
-import { addBabelPlugin, escapeRegExpCharacters, warnOnce } from './utils.js';
-import { RUNTIME, EXTERNAL, HELPERS } from './constants.js';
+import preflightCheck from './preflightCheck.js';
+import bundledHelpersPlugin from './bundledHelpersPlugin.js';
+import { addBabelPlugin, escapeRegExpCharacters } from './utils.js';
+import { RUNTIME, EXTERNAL, BUNDLED, INLINE, HELPERS } from './constants.js';
 
 const unpackOptions = ({
 	extensions = babel.DEFAULT_EXTENSIONS,
+	skipBabelHelpersCheck = false,
 	// rollup uses sourcemap, babel uses sourceMaps
 	// just normalize them here so people don't have to worry about it
 	sourcemap = true,
 	sourcemaps = true,
 	sourceMap = true,
 	sourceMaps = true,
 	...rest
-} = {}) => ({
-	extensions,
-	plugins: [],
-	sourceMaps: sourcemap && sourcemaps && sourceMap && sourceMaps,
-	...rest,
-	caller: {
-		name: 'rollup-plugin-babel',
-		supportsStaticESM: true,
-		supportsDynamicImport: true,
-		...rest.caller,
-	},
-});
+} = {}) => {
+	if (!rest.babelHelpers) {
```

Comment:
I'm tempted to say we should preserve some of the auto-detection functionality. It's super useful in two cases: 1) when you're getting started with rollup or a new project, and 2) for small projects where you want it to Just Work™. Larger projects definitely benefit from explicitly setting this.

One proposal: measure the number of times we run the preflight check for auto-detection and if it exceeds a certain number (or amount of time elapsed) then log a warning to the console asking the developer to specify the `babelHelpers` key (which they can choose to just be `auto` or something).

Reply:
My main motivation behind forcing this config on people was that people are often not aware that they "should" use runtime helpers.

I.e. since webpack@4 it requires "mode" config, however it defaults it to "production". Maybe we could do similarly here - leave current logic (no config, bundled helpers) as default BUT give a short warning (with a link to a full explanation) about configuring this explicitly. WDYT?

Reply:
Warning: wall of text incoming. I wanted to share most of my thought process, but see TLDR at bottom if you don't care :joy:

>My main motivation behind forcing this config on people was that people are often not aware that they "should" use runtime helpers.

Absolutely agreed.

I think in many ways forcing folks to configure this explicitly is a _really good thing_.

I'm kinda hoping to find a balance where it's possible to have a barebones config that invokes the `babel` plugin with no parameters when you're just getting started, and provides a really clear path towards a production-ready config.

Sometimes projects solve this with a bootstrap utility (`npm init`, for example). That could be an option, but IMO rollup's use-cases (mostly, source input and targets) are so varied that such a bootstrapping tool would be a very complex undertaking with too many moving parts to be useful.

I think I'm not in favor of a `mode`-type config item, but happy to be convinced otherwise. This is mostly on the grounds that I use rollup because it gives me a ton of power right out of the box, and doesn't make (too many) choices for me. This might inform our direction here, but I'm not sure if that's a common perspective.

>Maybe we could do similarly here - leave current logic (no config, bundled helpers) as default BUT give a short warning (with a link to a full explanation) about configuring this explicitly. WDYT?

I think this is more or less what I was attempting to say with "I'm tempted to say we should preserve some of the auto-detection functionality." If have the babel plugin run a preflight check to determine the appropriate `babelHelpers` key when not provided, and warn in that case, then I think that covers my concerns.

TLDR: :+1: on auto-detecting (no config, bundled helpers) as default with warning.